tests/virt_autotest/libvirt_virtual_network_init.pm

```diff
@@ -84,7 +84,14 @@ sub run_test {
         save_guest_ip($guest, name => "br123");
         virt_autotest::utils::ssh_copy_id($guest);
         #Prepare the new guest network interface files for libvirt virtual network
-        assert_script_run("ssh root\@$guest 'cd /etc/sysconfig/network/; cp ifcfg-eth0 ifcfg-eth1; cp ifcfg-eth0 ifcfg-eth2; cp ifcfg-eth0 ifcfg-eth3; cp ifcfg-eth0 ifcfg-eth4; cp ifcfg-eth0 ifcfg-eth5; cp ifcfg-eth0 ifcfg-eth6'");
+        #for some guests, interfaces are named eth0, eth1, eth2, ...
+        #for TW kvm guest, they are enp1s0, enp2s0, enp3s0, ...
+        my $primary_nic = script_output("ssh root\@$guest \"ip a|awk -F': ' '/state UP/ {print \\\$2}'|head -n1\"");
+        $primary_nic =~ /([a-zA-Z]*)(\d)(\w*)/;
+        for (my $i = 1; $i <= 6; $i++) {
+            my $nic = $1 . (int($2) + $i) . $3;
+            assert_script_run("ssh root\@$guest 'cd /etc/sysconfig/network/; cp ifcfg-$primary_nic ifcfg-$nic'");
+        }
```

Comment:
You are creating new interfaces from $primary_nic ?  What if $3 empty ?

Reply:
What about "cd ~" at the end ?

Reply:
> You are creating new interfaces from $primary_nic ? What if $3 empty ?

fg. $primary_nic='eth0',  $3 is empty,  the interfaces will be eth1, eth2, eth3, ...., eth6. There is a verification run with eth0: SLE kvm

Reply:
> What about "cd ~" at the end ?

Yes, better not to change the current work directory.

Reply:
@Julie-CAO Would you please explain the purpose of the for loop? Are you creating 6 new ifcfg-xx files? What are they for?

Reply:
yes, I created 6 new ifcfg-xx files. That's not what I did initially, I just adapted the test scenarios to TW from SLE. @guoxuguang , could you give more input about why we created 6 ifcfg-xx files in virtual network tests as you discussed with Pavel?

Reply:
What about wrap up this code snippet into a function ? create_network_interfaces_from_primary_nic or other name ? It takes the number of interfaces to be created as argument.

Reply:
To copy&paste numbers of network configuration files does not look like a decent way to implement the test, I prfer that this is a temporary and simple solution. So It is not necessary to make this immature code that is not needed elsewhere into a function.